MapboxNavigation/NetworkConfiguartion.swift

```diff
@@ -0,0 +1,31 @@
+import Foundation
+
+class NetworkConfiguration : NSObject{
+    @objc static let sharedConfiguration = NetworkConfiguration()
+    
+    // The current API base URL of map
+    private var apiBaseURL : String?
+    
+    // The PRC base URL for Mapbox APIs other than the telemetry API.
+    private let mapboxChinaBaseAPIURL = "https://api.mapbox.cn"
+    
+    // The base URL host for Mapbox China
+    public let mapboxChinaBaseURLHost = "api.mapbox.cn"
+    
+    // The URL String of China map style.
+    public let mapboxChinaStyleURL = "mapbox://styles/mapbox/streets-zh-v1"
+    
+    // Value of whether the map is China map or not
+    public var isChinaMap : Bool!{
+        guard apiBaseURL == mapboxChinaBaseAPIURL else {
+            return false
+        }
+        return true
+    }
+    
+    private override init() {
+        super.init()
+        apiBaseURL = Bundle.main.object(forInfoDictionaryKey:"MGLMapboxAPIBaseURL") as? String
```

Comment:
Is it necessary to store this base URL upfront? I think it would be more straightforward to execute this code on demand, as part of `hasChinaBaseURL`. It’s likely that Bundle already caches these info dictionary values.

Reply:
If you deep diving into our map SDK, this key have already load in +(void)load function in `MGLNetworkConfiguration.m`
![image](https://user-images.githubusercontent.com/2045318/42812565-9f7c1f92-89f0-11e8-8916-857567e5fc1f.png)
So that's NP we write here. Of course, if our map SDK didn't do this, we should execute this code on demand.

Reply:
Sure, the value doesn’t change, but my suggestion is to call `object(forInfoDictionaryKey:)` at the point of use anyways, because it shouldn’t affect performance significantly even if we call it several times in a row.